pkg/cmd/cli/cmd/logout.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	kclientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+	kcmdconfig "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/config"
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/templates"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+type LogoutOptions struct {
+	StartingKubeConfig *kclientcmdapi.Config
+	Config             *kclient.Config
+	Out                io.Writer
+
+	PathOptions *kcmdconfig.PathOptions
+}
+
+const logoutLongDescription = `Logs out the current user by deleting the token and removing the token from the kubeconfig file.
+`
+
+// NewCmdLogout implements the OpenShift cli logout command
+func NewCmdLogout(name, fullname string, f *osclientcmd.Factory, reader io.Reader, out io.Writer) *cobra.Command {
+	options := &LogoutOptions{
+		Out: out,
+	}
+
+	cmds := &cobra.Command{
+		Use:   name,
+		Short: "Logs out the current user.",
+		Long:  logoutLongDescription,
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(f, cmd, args); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+
+			if err := options.Validate(args); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+
+			if err := options.RunLogout(); err != nil {
+				kcmdutil.CheckErr(err)
+			}
+
+		},
+	}
+
+	templater := templates.Templater{
```

Comment:
Why do you need these? Unless you need to override the default behavior (i.e. explicitly exposing some global flag in help, like we do in `login`) you inherit the templates from parent command, so 58~62 should not be needed.